sfall/Modules/BarBoxes.cpp

```diff
@@ -177,59 +197,82 @@ void BarBoxes::SetText(int box, const char* text, DWORD color) {
 
 	ReconstructBarBoxes();
 
-	for (int i = 0; i < 5; i++) {
+	for (int i = 0; i < actualBoxCount; i++) {
 		boxesEnabled[i] = enabled[i];
 	}
+	delete[] enabled;
+
 	__asm call fo::funcoffs::refresh_box_bar_win_;
 }
 
 void BarBoxes::init() {
-	SafeWriteBatch<DWORD>((DWORD)boxes + 8, bboxMemAddr); //.mem
-	SafeWrite32(0x4612FE, (DWORD)boxes + 4); //.colour
-	SafeWrite32(0x46133C, (DWORD)boxes);     //.msg
 
-	int size = sSize * 10;
-	memset(boxes, 0, size);
-	memset(boxesEnabled, 0, 5 * 4);
+	boxCount = 5 + GetConfigInt("Misc", "BoxBarCount", 5);
+	if (boxCount < 5)   boxCount = 5;
+	if (boxCount > 120) boxCount = 120;
```

Comment:
Changed the set number of tags to the actual used, instead of the common value.

Reply:
I think boxCount should be 10 at minimum, just like in previous sfall builds there are always 5 extra boxes available.

Reply:
if return the previous version, there will be no consistency with the BoxBarColours option.
can be changed the condition, so that it is not possible to set it below 10.
boxCount = 5 + GetConfigInt("Misc", "BoxBarCount", 5);
if (boxCount < 10)   boxCount = 10;